sharding/nodesCoordinator/indexHashedNodesCoordinatorRegistry.go

```diff
@@ -61,7 +61,7 @@ func displayNodesConfigInfo(config map[uint32]*epochNodesConfig) {
 
 func (ihnc *indexHashedNodesCoordinator) saveState(key []byte, epoch uint32) error {
 	registry := ihnc.NodesCoordinatorToRegistry(epoch)
-	data, err := ihnc.nodesCoordinatorRegistryFactory.GetRegistryData(registry, ihnc.currentEpoch)
+	data, err := ihnc.nodesCoordinatorRegistryFactory.GetRegistryData(registry, epoch)
```

Comment:
were these 2 always equal?

Reply:
no, when executing EpochStartPrepare these are not equal. But then the state is saved again when these are equal, so it was still ok.